tests/unit/library/Episciences/User/Validate/EmailAvailabilityTest.php

```
@@ -0,0 +1,65 @@
+
+        self::assertFalse($validator->isValid('a@b.org'));
+    }
+
+    public function testMessagesAreResetBetweenCalls(): void
+    {
+        $validator = new EmailAvailability(0, EmailAvailability::DEFAULT_MESSAGE, static fn(string $email): array => [77]);
+
+        self::assertFalse($validator->isValid('a@b.org'));
+        self::assertNotEmpty($validator->getMessages());
+
+        $validator = new EmailAvailability(0, EmailAvailability::DEFAULT_MESSAGE, static fn(string $email): array => []);
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$email'. (undefined)

(UnusedFormalParameter)